src/main/java/ProcessManager.java

```diff
@@ -0,0 +1,290 @@
+        String number = line.substring(dividerPosition + TASK_DESCRIPTION_DIVIDER);
+        if (tryParse(number)) {
+            throw new DoneException("Done Request Has Invalid Number");
+        }
+        int numberInTaskArray = Integer.parseInt(number) - TASK_DESCRIPTION_DIVIDER;
+        if (getLastIndex() < numberInTaskArray || numberInTaskArray < ZERO) {
+            throw new DoneException("Index Of Task Does Not Exist");
+        }
+        taskList.get(numberInTaskArray).setIsDone();
+        String output = doneMessage(numberInTaskArray);
+        System.out.println(output);
+    }
+
+
+    public void handleListRequest() {
+        String output = LINE_DIVIDER;
+        for (int number = 0; taskList.size() > number; number++) {
+            String record = GAP + (number + ARRAY_INDEX_FINDER) + "." + taskList.get(number).toString() + "\n";
+            output = output.concat(record);
+        }
+        if (taskList.size() == ZERO) {
+            output = output.concat(GAP + "Nothing Has Been Added To The List Yet\n");
+        }
+        output = output.concat(LINE);
+        System.out.println(output);
+    }
+
+    public void handleDeleteRequest(String line) throws DeleteException {
+        if (line.length() == TASK_DATE_DIVIDER) {
+            throw new DeleteException("Request Does Not Contain A Number");
+        }
+        int dividerPosition = line.indexOf(DONE_DELETE_KEYWORD);
+        String number = line.substring(dividerPosition + TASK_DESCRIPTION_DIVIDER);
+        if (tryParse(number)) {
+            throw new DeleteException("Request Has Invalid Number");
+        }
+        int numberInTaskArray = Integer.parseInt(number) - TASK_DESCRIPTION_DIVIDER;
+        if (getLastIndex() < numberInTaskArray || numberInTaskArray < ZERO) {
+            throw new DeleteException("Array Out Of Bonds");
+        }
+        String output = deleteMessage(numberInTaskArray);
+        System.out.println(output);
+        taskList.remove(numberInTaskArray);
+    }
+
+    /* --- Function --- */
+    public boolean tryParse(String text) {
+        try {
+            Integer.parseInt(text);
+        } catch (NumberFormatException e) {
+            return true;
+        }
+        return false;
+    }
+
+    public int getLastIndex() {
+        int index = taskList.size() - ARRAY_INDEX_FINDER;
+        return Math.max(index, ZERO);
+    }
+
+    public void checkStatus(String status) throws DukeException {
+        if (status.equals("true")) {
+            taskList.get(getLastIndex()).setIsDone();
+        } else if (!status.equals("false")) {
+            throw new DukeException("Invalid Status");
+        }
+    }
+    /*--- -------- --- */
+
+    /*--- Messages --- */
+    public String taskAddedMessage() {
+        return LINE_DIVIDER + ADD_TASK_REPLY
+                + GAP + taskList.get(getLastIndex()).toString() + "\n"
+                + getAddTaskReturn(getLastIndex() + ARRAY_INDEX_FINDER) + LINE;
+    }
```

Comment:
Maybe it would be better to make such methods print the statements instead of adding a println in the function where method is called